isvcs/zk.go

```diff
@@ -232,7 +233,72 @@ func zkFourLetterWord(server, command string, timeout time.Duration) ([]byte, er
 	return resp, nil
 }
 
-func BuildZookeeperInstance(instanceID int) dao.RunningService{
+type ZooKeeperServerStats struct {
+	InstanceID  int
+	Connections int
+	Mode        string
+}
+
+func GetAllZooKeeperServerStats() []ZooKeeperServerStats {
+	stats := []ZooKeeperServerStats{}
+	for _, key := range GetZooKeeperKeys() {
+		stats = append(stats, GetZooKeeperServerStats(key.InstanceID, key.Connection))
+	}
+
+	return stats
+}
+
+func GetZooKeeperServerStatsByID(instanceID int) ZooKeeperServerStats {
+	for _, key := range GetZooKeeperKeys() {
+		if key.InstanceID != instanceID {
+			continue
+		}
+
+		return GetZooKeeperServerStats(instanceID, key.Connection)
+	}
+	return ZooKeeperServerStats{}
+}
+
+func GetZooKeeperServerStats(instanceID int, connection string) ZooKeeperServerStats {
+	logger := log.
+		WithField("instanceid", instanceID).
+		WithField("connection", connection)
+
+	stats := ZooKeeperServerStats{InstanceID: instanceID}
+
+	bytes, err := zkFourLetterWord(connection, "srvr", 1*time.Second)
+	if err != nil {
+		logger.WithError(err).Error("Unable to get ZooKeeper stats")
+		return stats
+	}
+
+	lines := strings.Split(string(bytes[:]), "\n")
+	for _, line := range lines {
+		if strings.HasPrefix(line, "Connections:") {
+			tokens := strings.SplitN(line, ":", 2)
+			value, err := strconv.Atoi(strings.TrimSpace(tokens[1]))
+			if err != nil {
+				logger.WithError(err).Error("Unable to get number of connections")
+			}
+			stats.Connections = value
+			logger = logger.WithField("connections", stats.Connections)
+			continue
+		}
+
+		if strings.HasPrefix(line, "Mode:") {
+			tokens := strings.SplitN(line, ":", 2)
+			if len(tokens) < 2 {
+				logger.Error("Unable to get mode")
+			}
+			stats.Mode = strings.TrimSpace(tokens[1])
+			logger = logger.WithField("mode", stats.Mode)
+		}
+	}
```

Comment:
If `stats` is empty after the loop, we should log either a Warning or Debug message so we might notice if a collection interval was missed.  Let's do a warning for now.